src/jvmMain/kotlin/Main.kt

```diff
@@ -32,8 +32,14 @@ import kotlin.io.path.inputStream
 
 @Composable
 fun App() {
-    var viewModel by rememberSaveable { mutableStateOf<DebuggerViewModel?>(null) }
+    val viewModel by rememberSaveable { mutableStateOf(DebuggerViewModel()) }
     var showFilePicker by remember { mutableStateOf(false) }
+    var pickedFile by remember { mutableStateOf<String?>(null) }
+
+    pickedFile?.let {
+        viewModel.loadJson(it)
+        pickedFile = null
+    }
```

Comment:
Add some comments to this "workaround" so we don't forget why it is there and then refactor it away and have to go through all your pain again.